app/service/knowledge_svc.py

```diff
@@ -37,12 +37,12 @@ async def update_fact(self, criteria, updates):
         # Update an existing fact
         return self.__loaded_knowledge_module._update_fact(criteria, updates)
 
-    async def get_facts(self, criteria):
+    async def get_facts(self, criteria, restrictions=None):
```

Comment:
So restrictions will be specified to these knowledge service methods? I thought the restrictions were inherent attributes of a fact? (like fact value can only be used once, or in current use once etc..)

(Please point me to updated spec/doc if this is noted somewhere)

Reply:
These restrictions are intended for the scoping of a fact to a host, network, etc. The facts themselves will still have a limit field for the number of times they can be used.